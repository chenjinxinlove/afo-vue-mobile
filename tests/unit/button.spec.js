import { shallowMount } from '@vue/test-utils'
import sinon from 'sinon'
import Vue from 'vue'
import Button from '../../src/button'
import Demo from '../../src/button/demo'

describe('Button.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Button)
    expect(Vue.component(Button.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 测试是否能正确的显示内容
  it('should render correct contents', () => {
    const wrapper = shallowMount(Button, {
      propsData: { icon: 'test' }
    })
    expect(wrapper.find('i').classes()).toContain('afo-test')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        effect: 'submit',
        type: 'primary',
        size: 'large',
        disabled: true,
        inline: true,
        plain: true
      }
    })
    expect(wrapper.attributes().type).toBe('submit')
    expect(wrapper.attributes().disabled).toBe('disabled')
    expect(wrapper.classes()).toEqual(['afo-button', 'afo-button--primaryplain', 'afo-button--large', 'afo-button-disabled', 'afo-button-inline'])
  })
  // 测试事件可以被处理
  it('events-called', () => {
    const clickHandler = sinon.stub()
    const wrapper = shallowMount(Button)
    wrapper.vm.$on('onClick', clickHandler)
    wrapper.trigger('click')
    expect(clickHandler.called).toBe(true)
  })
  // 测试事件disabled不被响应
  it('events-no-disabled-called', () => {
    const clickHandler = sinon.stub()
    const wrapper = shallowMount(Button, {
      propsData: {
        disabled: true
      }
    })
    wrapper.vm.$on('onClick', clickHandler)
    wrapper.trigger('click')
    expect(clickHandler.called).toBe(false)
  })
  // 测试事件loading不被响应
  it('events-no-loading-called', () => {
    const clickHandler = sinon.stub()
    const wrapper = shallowMount(Button, {
      propsData: {
        loading: true
      }
    })
    wrapper.vm.$on('onClick', clickHandler)
    wrapper.trigger('click')
    expect(clickHandler.called).toBe(false)
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(shallowMount(Demo).html()).toMatchSnapshot()
  })
})
