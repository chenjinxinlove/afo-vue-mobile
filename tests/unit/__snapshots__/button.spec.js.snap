// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue demo snapshot 1`] = `
<div>
  <div>
    <p>按钮用途</p>
    <div class="show">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <!---->
        <!---->默认按钮</button>
      <button type="submit" class="afo-button afo-button--default afo-button--normal">
        <!---->
        <!---->提交按钮</button>
    </div>
  </div>
  <div>
    <p>按钮类型</p>
    <div class="show">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <!---->
        <!---->默认按钮</button>
      <button type="button" class="afo-button afo-button--primary afo-button--normal">
        <!---->
        <!---->主要按钮</button>
      <button type="button" class="afo-button afo-button--danger afo-button--normal">
        <!---->
        <!---->危险按钮</button>
    </div>
  </div>
  <div>
    <p>素按钮类型</p>
    <div class="show">
      <button type="button" class="afo-button afo-button--defaultplain afo-button--normal">
        <!---->
        <!---->默认按钮</button>
      <button type="button" class="afo-button afo-button--primaryplain afo-button--normal">
        <!---->
        <!---->主要按钮</button>
      <button type="button" class="afo-button afo-button--dangerplain afo-button--normal">
        <!---->
        <!---->危险按钮</button>
    </div>
  </div>
  <div>
    <p>按钮大小</p>
    <div class="show marginbottom">
      <button type="button" class="afo-button afo-button--default afo-button--normal">
        <!---->
        <!---->默认按钮</button>
      <button type="button" class="afo-button afo-button--default afo-button--small">
        <!---->
        <!---->小号按钮</button>
    </div>
    <button type="button" class="afo-button afo-button--default afo-button--large">
      <!---->
      <!---->大号按钮</button>
  </div>
  <div>
    <p>禁用状态</p>
    <button type="button" disabled="disabled" class="afo-button afo-button--default afo-button--normal afo-button-disabled">
      <!---->
      <!---->禁用状态</button>
  </div>
  <div>
    <p>行内按钮</p>
    <button type="button" class="afo-button afo-button--default afo-button--normal afo-button-inline">
      <!---->
      <!---->行内按钮</button>
  </div>
  <div>
    <p>loading按钮</p>
    <button type="button" class="afo-button afo-button--default afo-button--normal">
      <!---->
      <div class="afo-loading afo-loading-black"><span class="afo-loading-spinners" style="width: 24px; height: 24px;"><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i><i class="afo-loading-spinner"></i></span></div>
      loading按钮</button>
  </div>
  <div>
    <p>Icon按钮</p>
    <button type="button" class="afo-button afo-button--default afo-button--normal">
      <i class="afo-icon-left"></i>
      <!---->Icon按钮</button>
  </div>
</div>
`;
